packages/format-library/src/link/index.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { withSpokenMessages } from '@wordpress/components';
+import {
+	getTextContent,
+	applyFormat,
+	removeFormat,
+	slice,
+} from '@wordpress/rich-text';
+import { isURL } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import InlineLinkUI from './inline';
+
+const name = 'core/link';
+
+export const link = {
+	name,
+	title: __( 'Link' ),
+	match: {
+		tagName: 'a',
+	},
+	attributes: {
+		url: 'href',
```

Comment:
I thought this is something that could change it the future, so I went for the map. Maybe an attribute could be sourced from part, or several, element attributes, and maybe in the future we'll support more complex objects.

Reply:
>>For what purpose do we support attributes defined as a mapping.
>
>I thought this is something that could change it the future, so I went for the map. Maybe an attribute could be sourced from part, or several, element attributes, and maybe in the future we'll support more complex objects.

Well, that was the sort of thing I'd first expected when looking at this, since it _appears_ to mimic the block API for attributes. But what we have here seems more like a confusing middle ground. Why not just have the full verbose attribute definition which matches completely the expectation of the block API? Or, if that's not intended to be supported yet, have the simpler array alternative as the short-term available option, since it's not incompatible with such a future proposed enhancement.

Reply:
That’s, fine with me as well. I’ll make a separate or for it